File: day_2/part_1.py
file = open("input.txt")
text = file.read().splitlines()[0]
print("Day 2: Part one")

def valid(number):
    number = str(number)
    if len(number) % 2 != 0:
        return True
    n = len(number)
    firstHalf = number[:n//2]
    secondHalf = number[n//2:]
    return not(firstHalf == secondHalf)



def sum_invalid(low, high):
    out = 0 
    for x in range(low, high+1):
        if not valid(x):
            out += x
    return out


total = 0 

for interval in text.split(","):
    begin, end = interval.split("-")
    begin = int(begin)
    end = int(end)
    total += sum_invalid(begin, end)


print("Total: ", total)

File: day_5/src/part_1.rs
use std::fs;


pub fn main(){
    let input_str = fs::read_to_string("input.txt").expect("Input file not found!");
    let mut intervals: Vec<(u64, u64)> = Vec::new();
    let mut past_half = false;

    let mut total = 0;


    for line in input_str.split("\n"){
        if line.is_empty(){
            past_half = true;
            continue;
        }
        if !past_half{
            let mut line_split = line.split("-");
            let low = line_split.next().expect("Not enough values in interval (low)!");
            let high = line_split.next().expect("Not enough values in interval (high)!");
            intervals.push((low.parse::<u64>().expect("Not an integer!"), high.parse::<u64>().expect("Not an integer (intervals)!")));
        }else{
            let num: u64 = line.parse().expect("Not an integer (num)!");
            for interval in intervals.iter(){
                if num >= interval.0 && num <= interval.1{
                    total += 1;
                    break;
                }
            }
        }
    }


    println!("Total number of fresh ingredients: {}", total);
}
